pkg/install/app/vaultconfigdeps.go

```diff
@@ -0,0 +1,239 @@
+package app
+
+import (
+	"context"
+	"strings"
+
+	batchv1obj "github.com/puppetlabs/leg/k8sutil/pkg/controller/obj/api/batchv1"
+	corev1obj "github.com/puppetlabs/leg/k8sutil/pkg/controller/obj/api/corev1"
+	"github.com/puppetlabs/leg/k8sutil/pkg/controller/obj/helper"
+	"github.com/puppetlabs/leg/k8sutil/pkg/controller/obj/lifecycle"
+	"github.com/puppetlabs/relay-core/pkg/apis/install.relay.sh/v1alpha1"
+	"github.com/puppetlabs/relay-core/pkg/model"
+	"github.com/puppetlabs/relay-core/pkg/obj"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	vaultAddr           = "http://vault:8200"
+	vaultIdentifier     = "vault"
+	vaultInitIdentifier = "vault-init"
+
+	vaultAddrEnvVar           = "RELAY_OPERATOR_VAULT_INIT_VAULT_ADDR"
+	vaultJWTAuthPathEnvVar    = "RELAY_OPERATOR_VAULT_INIT_VAULT_JWT_AUTH_PATH"
+	vaultJWTMountEnvVar       = "RELAY_OPERATOR_VAULT_INIT_VAULT_JWT_MOUNT"
+	vaultJWTPublicKeyEnvVar   = "RELAY_OPERATOR_VAULT_INIT_VAULT_JWT_PUBLIC_KEY"
+	vaultNameEnvVar           = "RELAY_OPERATOR_VAULT_INIT_VAULT_NAME"
+	vaultNamespaceEnvVar      = "RELAY_OPERATOR_VAULT_INIT_VAULT_NAMESPACE"
+	vaultServiceAccountEnvVar = "RELAY_OPERATOR_VAULT_INIT_VAULT_SERVICE_ACCOUNT"
+	vaultTokenEnvVarName      = "RELAY_OPERATOR_VAULT_INIT_VAULT_TOKEN"
+	vaultUnsealKeyEnvVarName  = "RELAY_OPERATOR_VAULT_INIT_VAULT_UNSEAL_KEY"
+
+	logServicePathEnvVar            = "RELAY_OPERATOR_VAULT_INIT_LOG_SERVICE_PATH"
+	logServiceVaultAgentRoleEnvVar  = "RELAY_OPERATOR_VAULT_INIT_LOG_SERVICE_VAULT_AGENT_ROLE"
+	metadataAPIVaultAgentRoleEnvVar = "RELAY_OPERATOR_VAULT_INIT_METADATA_API_VAULT_AGENT_ROLE"
+	operatorVaultAgentRoleEnvVar    = "RELAY_OPERATOR_VAULT_INIT_OPERATOR_VAULT_AGENT_ROLE"
+	tenantPathEnvVar                = "RELAY_OPERATOR_VAULT_INIT_TENANT_PATH"
+	transitKeyEnvVar                = "RELAY_OPERATOR_VAULT_INIT_TRANSIT_KEY"
+	transitPathEnvVar               = "RELAY_OPERATOR_VAULT_INIT_TRANSIT_PATH"
+)
+
+type VaultConfigDeps struct {
+	Core                *obj.Core
+	ConfigJob           *batchv1obj.Job
+	JWTSigningKeySecret *corev1obj.Secret
+	OwnerConfigMap      *corev1obj.ConfigMap
+	Labels              map[string]string
+}
+
+func (vd *VaultConfigDeps) Load(ctx context.Context, cl client.Client) (bool, error) {
+	if _, err := vd.Core.Load(ctx, cl); err != nil {
```

Comment:
In general, I think there's assumptions made in this code that if you pass a full object (like `Core`, here) into the constructor of something else, it's already been loaded and you don't need to do it again, although it doesn't really hurt anything (will probably just be a cache hit).

Reply:
I'm following the existing pattern here. `LogServiceDeps`, 'MetadataAPIDeps`, `OperatorDeps`, `VaultAgentDeps`, and `WebhookCertificateControllerDeps` are all doing this currently. I'll leave this as-is for now, and evaluate later. If we can remove all of these calls, cool.